src/Utilities/Compiler/CodeMetrics/MetricsHelper.cs

```diff
@@ -265,21 +265,30 @@ static bool hasConditionalLogic(IOperation operation)
         private static void AddCoupledNamedTypesCore(ImmutableHashSet<INamedTypeSymbol>.Builder builder, ITypeSymbol typeOpt)
         {
             if (typeOpt is INamedTypeSymbol usedType &&
-                !isIgnoreableType(usedType) &&
-                builder.Add(usedType))
+                !isIgnoreableType(usedType))
             {
-                if (usedType.IsGenericType)
+                if (usedType.IsGenericType && !usedType.IsUnboundGenericType)
                 {
+                    builder.Add(usedType.ConstructUnboundGenericType());
```

Comment:
@mavasani will correct me but I think that's better to use the `OriginalDefinition` property. You can use it in both the if and else as it will return the same value if there is no generics.

Reply:
Ping @j2jensen

Reply:
Sorry it has taken me a while. I haven't forgotten this, just haven't made time for it. I should have a chance to work on it again this Thursday.

Reply:
Changed to use OriginalDefinition.